src/Blueprint/Blueprint/Models/PSBlueprint.cs

```diff
@@ -0,0 +1,120 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Management.Blueprint.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Microsoft.Azure.PowerShell.Cmdlets.Blueprint.Common;
+using System.Collections;
+
+namespace Microsoft.Azure.Commands.Blueprint.Models
+{
+    public class PSBlueprint : PSBlueprintBase
+    {
+        public string DefinitionLocationId { get; set; }
+        public string Scope { get; set; }
+        public object Versions { get; set; }
+        public List<string> ParametersDisplayList { get; set; }
+        public List<string> ResourceGroupDisplayList { get; set; }
+        public List<string> VersionList { get; set; }
+
+        /// <summary>
+        /// Create a PSBlueprint object from a BlueprintModel.
+        /// </summary>
+        /// <param name="model"></param>
+        /// <param name="managementGroupName">Name of the management group the blueprint belongs to.</param>
+        /// <returns>A new PSBlueprint object</returns>
+        internal static PSBlueprint FromBlueprintModel(BlueprintModel model, string scope)
+        {
+            var psBlueprint = new PSBlueprint
+            {
+                Id = model.Id,
+                Name = model.Name,
+                DefinitionLocationId = Utils.GetDefinitionLocationId(scope),
+                Scope = scope,
+                DisplayName = model.DisplayName,
+                Description = model.Description,
+                Status = new PSBlueprintStatus(),
+                TargetScope = PSBlueprintTargetScope.Unknown,
+                Parameters = new Dictionary<string, PSParameterDefinition>(),
+                ResourceGroups = new Dictionary<string, PSResourceGroupDefinition>(),
+                Versions = model.Versions,
+                ParametersDisplayList = new List<string>(),
+                ResourceGroupDisplayList = new List<string>(),
+                VersionList = new List<string>()
+            };
+
+            if (DateTime.TryParse(model.Status.TimeCreated, out DateTime timeCreated))
```

Comment:
>TryParse [](start = 25, length = 8)

If it fails to parse, we silently skip assinging the value. That probably means TimeCreated set to Datetime.MinValue. Do we want that?

Reply:
Same for all the TryParse calls here.

---
In reply to: [257121303](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257121303) [](ancestors = 257121303)